File: src/main/java/ua/nure/kn/zhmaka/usermanagement/db/UserDAO.java
package ua.nure.kn.zhmaka.usermanagement.db;

import java.util.Collection;

import ua.nure.kn.zhmaka.usermanagement.User;

public interface UserDAO {

    User create(User user) throws DatabaseException;
    void update(User user) throws DatabaseException;
    void delete(Long id) throws DatabaseException;
    
    User find(Long id) throws DatabaseException;
    Collection findAll() throws DatabaseException;

    public ConnectionFactory getConnectionFactory();

    public void setConnectionFactory(ConnectionFactory connectionFactory);
}

File: src/main/kotlin/ua/nure/kn/zhmaka/usermanagement/User/DAO/UserDao.kt
package ua.nure.kn.zhmaka.usermanagement.User.DAO

import org.jetbrains.exposed.sql.SizedIterable
import ua.nure.kn.zhmaka.usermanagement.User.models.User
import ua.nure.kn.zhmaka.usermanagement.User.models.UserPOJO

//define a basic CRUD operations
interface UserDao {
    fun save(u : UserPOJO):User

    fun find(id :Int):User?

    fun findAll():SizedIterable<User>

    fun update(u:User)

    fun delete(id: Int)
}